Add downsampling debug reports section to explainer. (#1020)
WICG/turtledove: FLEDGE.md

#### 6.4 Reporting Additional Bid Wins

For additional bids that win the auction, event-level win reporting is supported, just as it is for bids generated from stored interest groups. However, additional bids have distinct security concerns. The browser can guarantee that bids generated from stored interest groups were, in fact, generated by the buyer's origin, but it cannot provide this same guarantee for additional bids. An additional bid can only win if none of the negative interest groups specified by the additional bid are present on the user's device, and so the signature validation described above in section [6.2.3 Additional Bid Keys](#623-additional-bid-keys) also can't help with this. To ensure that buyers understand that they're being asked to report on an additional bid, which cannot be verified having been generated by the buyer's origin, the reporting for additional bids is done by calling reporting functions with a slightly different name. Instead of `reportWin()`, the browser instead calls a function named `reportAdditionalBidWin()`to report a winning additional bid.

### 7. Debugging Extensions

> 

Reviewer: ```suggestion

#### 7.1 forDebuggingOnly (fDO) APIs

In order to accomplish our dual goals of helping with adoption and preserving user privacy, we plan to keep the forDebuggingOnly APIs available post third-party cookie deprecation, albeit in a heavily downsampled fashion (as described in [section 7.1.1](#711-downsampling)) to address the re-identification risk.  While third-party cookies are available, they will remain unsampled to help with adoption, and instead will have a label, exposed in `generateBid()`'s `browserSignals.forDebuggingOnlyInCooldownOrLockout`, to indicate whether the particular report would have been downsampled.

```

#### 7.1.1 Downsampling

> 

Reviewer: How about moving this up to section 7.1 and replacing "the PR" with "section 7.1.1"?

> 

Reviewer 2: done

This design has two main goals:
*   Prevent sending fDO reports very often to protect user privacy:
Only sending reports 1/1000 times the forDebuggingOnly API is called, and
If a report is sent, “lock-out” all adtechs out of sending a report for 3 years, and
If an adtech calls the API, put them in a  “cool-down” period where calls to the API by that given adtech are not able to send reports.

> *  Prevent adtechs who accidentally (e.g. due to a bug in their code) call the API repeatedly for all users, from locking themselves out of sending any more reports for years. This is accomplished by 90% of the time putting that adtech in a 2 week “cooldown” period, and only 10% of the time putting that adtech in a 1 year cooldown period.

Author: maybe useful to add a link to the github issue for how we calculated the numbers?

Reviewer: Sure, perhaps you could add a sentence at the end like "See [issue 632](https://github.com/WICG/turtledove/issues/632) for more information."

    *   "language-exclusions"
    *   "category-exclusions"

#### 7.1.2 Downsampling

> This design has two main goals:

Reviewer 3: ```suggestion
This design has three main goals:
```

*   Prevent sending fDO reports very often to protect user privacy:

> Only sending reports 1/1000 times the forDebuggingOnly API is called, and

Reviewer 3: ```suggestion
  * Only sending reports 1/1000 times the forDebuggingOnly API is called, and
```

> If a report is sent, “lock-out” all adtechs out of sending a report for 3 years, and

Reviewer 3: ```suggestion
  * If a report is sent, “lock-out” all adtechs out of sending a report for 3 years, and
```

Reviewer 3: (Could you also replace the smart-quote characters, “ and ”, with regular quote character ", throughout?)

Author: Done. Thanks for noticing!

> If an adtech calls the API, put them in a  “cool-down” period where calls to the API by that given adtech are not able to send reports.

Reviewer 3: ```suggestion
  * If an adtech calls the API, put them in a  “cool-down” period where calls to the API by that given adtech are not able to send reports.
```

> *  Prevent adtechs who accidentally (e.g. due to a bug in their code) call the API repeatedly for all users, from locking themselves out of sending any more reports for years. This is accomplished by 90% of the time putting that adtech in a 2 week “cooldown” period, and only 10% of the time putting that adtech in a 1 year cooldown period.

Reviewer 3: ```suggestion
*  Prevent one adtech from substantially compromising a different adtech's access to debugging information.  The "cooldown" period means that any particular adtech can only cause a very small fraction of people to send debug reports and be removed from the potential debugging pool.
*  Prevent adtechs who accidentally (e.g. due to a bug in their code) call the API repeatedly for all users, from locking themselves out of sending any more reports for years. This is accomplished by 90% of the time putting that adtech in a 2 week cooldown period, and only 10% of the time putting that adtech in a 1 year cooldown period.
```

See [issue 632](https://github.com/WICG/turtledove/issues/632) for more information.

> Both `generateBid()` and `scoreAd()`’s `browserSignals` have a new boolean field `forDebuggingOnlyInCooldownOrLockout`. It’s true when adtechs are in “lock-out” period, or when the buyer (or seller of `scoreAd()`) is in “cool-down” period.

Reviewer 3: ```suggestion
Both `generateBid()` and `scoreAd()`’s `browserSignals` have a new boolean field `forDebuggingOnlyInCooldownOrLockout`. It’s true when the API is in the global “lock-out” period, or when this particular buyer or seller, respectively, is in “cool-down” period.
```

    *   "language-exclusions"
    *   "category-exclusions"

#### 7.1.2 Downsampling

This design has three main goals:

> *   Prevent sending fDO reports very often to protect user privacy:

Reviewer 3: ```suggestion
* Prevent sending fDO reports very often to protect user privacy:
```

Reviewer 3: (Oops sorry, I think this will make the next three bullets properly nest under this one)
